### PR TITLE
chore(install): allow non us regions, and add more debug statements. …

### PR DESCRIPTION
**Pull request by Julien4218**
…Fix issue when recipe is not executed with -c

### CODE REVIEW

File: internal/install/recipe_installer.go [hunk 1]
```diff
@@ -123,36 +130,57 @@ func (i *RecipeInstaller) Install() error {
...
 
 		for _, r := range recipes {
-			log.Debugf("Found available integration %s.", r.Name)
+			log.Debugf(fmt.Sprintf("Found available integration %s.", r.Name))
 		}
 
 		i.reportRecipesAvailable(recipes)
 	}
 
+	log.Debugf(fmt.Sprintf("AssumeYes:%t.", i.AssumeYes))
```
Comment (ctrombley) [inline internal/install/recipe_installer.go:139]: This multi-statement debug block can also be achieved with `log.Debugf(%+v\n.", i.InstallerContext)`, but with slightly harder-to-read format.

File: internal/install/recipe_installer.go [hunk 2]
```diff
@@ -178,13 +206,18 @@ func (i *RecipeInstaller) Install() error {
 
 func (i *RecipeInstaller) installRecipesWithPrompts(m *types.DiscoveryManifest, recipes []types.Recipe, entityGUID string) (bool, error) {
 	errorsEncountered := false
+	log.Debugf("Installing recipes with prompts...")
 
 	for _, r := range recipes {
+		log.Debugf(fmt.Sprintf("Installing recipe %s with prompts...", r.Name))
```
Comment (ctrombley) [inline internal/install/recipe_installer.go:212]: `fmt.Sprintf` is unnecessary from within `log.Debugf` - you can use it the same way you would `Sprintf`.

File: internal/install/recipe_installer.go [hunk 3]
```diff
@@ -194,29 +227,36 @@ func (i *RecipeInstaller) installRecipesWithPrompts(m *types.DiscoveryManifest,
 		if i.RecipesProvided() || i.AssumeYes {
 			ok = true
 		} else {
+			log.Debugf("Checking user accepts install...")
 			ok, err = i.userAcceptsInstall(r)
 			if err != nil {
+				log.Debugf(fmt.Sprintf("Done installing recipes with prompts, exception:%s", err))
```
Comment (ctrombley) [inline internal/install/recipe_installer.go:233]: fmt.Sprintf is unnecessary from within log.Debugf - you can use it the same way you would Sprintf.

File: internal/install/recipe_installer.go [hunk 4]
```diff
@@ -194,29 +227,36 @@ func (i *RecipeInstaller) installRecipesWithPrompts(m *types.DiscoveryManifest,
...
 				return true, err
 			}
+			log.Debugf(fmt.Sprintf("Done checking user accepts install ok:%t", ok))
 		}
 
 		if !ok {
-			log.Debugf("skipping recipe %s.", r.Name)
+			log.Debugf(fmt.Sprintf("skipping not ok recipe %s.", r.Name))
```
Comment (ctrombley) [inline internal/install/recipe_installer.go:240]: fmt.Sprintf is unnecessary from within log.Debugf - you can use it the same way you would Sprintf.

File: internal/install/recipe_installer.go [hunk 5]
```diff
@@ -194,29 +227,36 @@ func (i *RecipeInstaller) installRecipesWithPrompts(m *types.DiscoveryManifest,
 		if i.RecipesProvided() || i.AssumeYes {
 			ok = true
 		} else {
+			log.Debugf("Checking user accepts install...")
 			ok, err = i.userAcceptsInstall(r)
 			if err != nil {
+				log.Debugf(fmt.Sprintf("Done installing recipes with prompts, exception:%s", err))
 				return true, err
 			}
+			log.Debugf(fmt.Sprintf("Done checking user accepts install ok:%t", ok))
```
Comment (ctrombley) [inline internal/install/recipe_installer.go:236]: fmt.Sprintf is unnecessary from within log.Debugf - you can use it the same way you would Sprintf.

File: internal/install/recipe_installer.go (hunk 1 above)
Comment (Julien4218) [reply internal/install/recipe_installer.go:139]: Nice, I was wishing for something like that, I'll update. I don't mind that it is harder to read in debug. It's also more maintainable if more fields are added in the future.

File: internal/install/recipe_installer.go (hunk 2 above)
Comment (Julien4218) [reply internal/install/recipe_installer.go:212]: I was hoping for that, but it's not what I'm noticing when it runs with just the log.Debugf( ). In the output, I see the %s in the message, and the parameter added after the entire string.

File: internal/install/recipe_installer.go (hunk 1 above)
Comment (Julien4218) [reply internal/install/recipe_installer.go:139]: Ho my bad, I've mistaken log.DebugLn with log.Debugf.
Do I need to explicit do a line return if I use log.Debugf ?

File: internal/install/recipe_installer.go (hunk 2 above)
Comment (Julien4218) [reply internal/install/recipe_installer.go:212]: Ho my bad, I've mistaken log.DebugLn with log.Debugf.
Do I need to explicit do a line return if I use log.Debugf ?

**Review by ctrombley — APPROVED** [review]
:shipit:

File: internal/install/recipe_installer.go [hunk 6]
```diff
@@ -92,29 +92,36 @@ func (i *RecipeInstaller) Install() error {
 	if i.RecipePathsProvided() {
 		// Load the recipes from the provided file names.
 		for _, n := range i.RecipePaths {
+			log.Debugln(fmt.Sprintf("Attempting to match recipePath %s.", n))
```
Comment (zlesnr) [inline internal/install/recipe_installer.go:95]: `Debugf` is a good way to combine this into smaller code.  `log.Debugf("Attempting to match recipePath %s.", n)` is a little less to look at

**Review by zlesnr — COMMENTED** [review]
Thanks for this.

**Pull request merged**